File: arrays-and-hashing/two-sum.py
def twoSum(nums, target):
     """
     Finds two numbers in the given list that add up to the target value.

     Args:
          nums (list): A list of integers.
          target (int): The target value.

     Returns:
          list: A list containing the indices of the two numbers that add up to the target value.
     """
     hashMap = {}
     
     for indx, val in enumerate(nums):
          valueWeNeed = target - val
          if (valueWeNeed in hashMap):
               return [indx, hashMap[valueWeNeed]]
          hashMap[val] = indx
     
     return


nums = [2,7,11,15]
target = 9

print(twoSum(nums, target))

File: arrays-and-hashing/java-solutions/TwoSum.java
import java.util.*;

public class TwoSum {
     public static void main(String[] args) {
          int[] indices = twoSum(new int[]{1,2,3,4}, 5);
          for (int index : indices) {
               System.out.println(index);
          }
          return;
     }

     public static int[] twoSum(int[] nums, int target) {
          HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
          
          for (int i = 0; i < nums.length; i++) {
               int valWeNeed = target - nums[i];
               if (map.containsKey(valWeNeed)) {
                    return new int[]{map.get(valWeNeed), i};
               }
               map.put(nums[i], i);
          }
          return new int[]{0};
     }
}
